pilot/pkg/model/service.go

```diff
@@ -1016,6 +1024,12 @@ type ServiceInfo struct {
 	// Source is the type that introduced this service.
 	Source   TypedObject
 	Waypoint WaypointBindingStatus
+	// MarshaledAddress contains the pre-marshaled representation.
+	// Note: this is an Address -- not a Service.
+	MarshaledAddress *anypb.Any
```

Comment:
nit: do these two really need to exported? If nothing outside of this package needs to set these two (seems weird to me that we might?) and we're marshaling `ServiceInfo`, could we just memoize the data in private fields and create accessor methods? Not blocking

Reply:
They are set in the ambient package so I think so?

Reply:
Ah I somehow missed that this was in pkg/model - disregard